### ISSUE TITLE
[BottomAppBar] How to get current scroll state (STATE_SCROLLED_UP | STATE_SCROLLED_DOWN) ? 

### ISSUE THREAD

**Issue by dariush-fathie** [material-components/material-components-android#2185]
There is no way to find that whether `BottomAppBar` is scrolled down or not
Even the `BottomAppBar.behavior.currentState` is private field

It's will be good to have scroll state callback and BottomAppBar.isScrolledDown() method
```kotlin
BottomAppBar.addScrollStateCallback(object : BottomAppBar.ScrollStateCallback() {
      override fun onStateChanged(bottomAppBar: View, newState: Int) {
      }

      override fun onSlide(bottomAppBar: View, slideOffset: Float) {
      }
   })

BottomAppBar.isScrolledDown(): Boolean
```

**Comment by wax911** [comment]
Agreed, I have been trying to get the current state as well but no real public interface exists, I was thinking of something a lot simpler. Since we already have methods like `performHide()` and `performShow()`, having `isShown` or `isHidden` would be a good place to start

**Comment by imhappi** [comment]
Fixed by [21f9b0ff25d23506e9300f057a42bd24508d1b7e](https://github.com/material-components/material-components-android/commit/21f9b0ff25d23506e9300f057a42bd24508d1b7e)
